### PR TITLE
Add diligent-core/2.5.2

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **diligent-core/2.5.2**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
implements https://github.com/conan-io/conan-center-index/issues/10089

**Comment by AndreyMlashkin** [comment]
I see, that this file exists in glslang package: ./glslang/1.3.204.0/_/_/package/1b40aaa558973b5b615e216fa40051e40fb1eba1/include/glslang/SPIRV/GlslangToSpv.h

But it's not in the includepath. What's the best way to fix it? Add it in conanfile or do some cmake file patch?
Or maybe add 'include/glslang/' to includepatch in glslang recipe?

**Comment by AndreyMlashkin** [comment]
> I see, that this file exists in glslang package: ./glslang/1.3.204.0/_/_/package/1b40aaa558973b5b615e216fa40051e40fb1eba1/include/glslang/SPIRV/GlslangToSpv.h
> 
> But it's not in the includepath. What's the best way to fix it? Add it in conanfile or do some cmake file patch? Or maybe add 'include/glslang/' to includepatch in glslang recipe?
> 
> @SpaceIm @SSE4 @uilianries

@ericLemanissier @madebr @intelligide 
need some help here

**Comment by AndreyMlashkin** [comment]
Yes, it is a public include.
And yes, that is the question, if I should use "include "glslang/SPIRV/GlslangToSpv.h"

**Comment by SpaceIm** [comment]
> > I see, that this file exists in glslang package: ./glslang/1.3.204.0/_/_/package/1b40aaa558973b5b615e216fa40051e40fb1eba1/include/glslang/SPIRV/GlslangToSpv.h
> > But it's not in the includepath. What's the best way to fix it? Add it in conanfile or do some cmake file patch? Or maybe add 'include/glslang/' to includepatch in glslang recipe?
> 
> I am not familar with `gslang` or `diligent`. is it a public include? if so, it should be added into `includepath` in `gslang` conanfile. also, is it supposed to be included as:
> 
> ```
> #include "SPIRV/GlslangToSpv.h"
> ```
> 
> or
> 
> ```
> include "glslang/SPIRV/GlslangToSpv.h"
> ```

It's

```
include "glslang/SPIRV/GlslangToSpv.h"
```

**Comment by AndreyMlashkin** [comment]
Do you know, is there a better way to fix it on cmake level? It maybe there are more such includes

**Review by intelligide — APPROVED** [review]

File: recipes/diligent-core/all/patches/0016-252-fix-glslang-usage.patch [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+index c2dd85f3..cc83e707 100644
+--- a/Graphics/ShaderTools/CMakeLists.txt
++++ b/Graphics/ShaderTools/CMakeLists.txt
+@@ -150,8 +150,8 @@ if(ENABLE_SPIRV)
+     if (${USE_GLSLANG})
+         target_link_libraries(Diligent-ShaderTools
+         PRIVATE
+-            SPIRV
+-            glslang
++            spirv-tools::SPIRV-Tools-opt
```
Comment (SpaceIm) [inline recipes/diligent-core/all/patches/0016-252-fix-glslang-usage.patch:11]:
```suggestion
+            SPIRV-Tools-opt
```

File: recipes/diligent-core/all/conanfile.py [hunk 2]
```diff
@@ -91,7 +93,8 @@ def requirements(self):
 
         self.requires("spirv-cross/1.3.204.0")
         self.requires("spirv-tools/1.3.204.0")
-        self.requires("glslang/1.3.204.0")
+        if self.options.with_glslang:
+            self.requires("glslang/1.3.204.0")
         self.requires("vulkan-headers/1.3.204.0")
```
Comment (SpaceIm) [inline recipes/diligent-core/all/conanfile.py:98]:
```suggestion
        self.requires("vulkan-headers/1.3.204.1")
```

**Review by uilianries — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
@jgsogo @SpaceIm @SSE4
please approve. It is already 10 days in review

**Review by SSE4 — APPROVED** [review]

**Pull request merged**